modules/ecs-cluster-ec2-asg/v0.0.1/main.tf

```diff
@@ -0,0 +1,308 @@
+# Create KMS key
+resource "aws_kms_key" "this" {
+  description             = "ECS Cluster ${var.cluster_name}"
+  deletion_window_in_days = 7
+  enable_key_rotation     = true
+  tags                    = var.tags
+}
+
+resource "aws_kms_alias" "this" {
+  name          = "alias/ecs/cluster/${var.cluster_name}"
+  target_key_id = aws_kms_key.this.key_id
+}
+
+resource "aws_kms_key_policy" "this" {
+  key_id = aws_kms_key.this.id
+  policy = jsonencode({
+    Id = "KMSKeyPolicy"
+    Statement = [
+      {
+        Action = "kms:*"
+        Effect = "Allow"
+        Principal = {
+          AWS = data.aws_caller_identity.current.account_id
+        }
+        Resource = "*"
+        Sid      = "Enable IAM User Permissions"
+      },
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "logs.${data.aws_region.current.name}.amazonaws.com"
+        },
+        "Action" : [
+          "kms:Encrypt*",
+          "kms:Decrypt*",
+          "kms:ReEncrypt*",
+          "kms:GenerateDataKey*",
+          "kms:Describe*"
+        ],
+        "Resource" : "*",
+        "Condition" : {
+          "ArnLike" : {
+            "kms:EncryptionContext:aws:logs:arn" : "arn:aws:logs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:log-group:/ECS/Cluster/${var.cluster_name}"
+          }
+        }
+      },
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "ssm.amazonaws.com"
+        },
+        "Action" : [
+          "kms:Encrypt*",
+          "kms:Decrypt*",
+          "kms:ReEncrypt*",
+          "kms:GenerateDataKey*",
+          "kms:Describe*"
+        ],
+        "Resource" : "*"
+      },
+    ]
+    Version = "2012-10-17"
+  })
+}
+
+resource "aws_cloudwatch_log_group" "this" {
+  #checkov:skip=CKV_AWS_338:CloudWatch log groups log retention will be set to the desired period
+  depends_on = [
+    aws_kms_key_policy.this
+  ]
+  name              = "/ECS/Cluster/${var.cluster_name}"
+  retention_in_days = var.cloudwatch_log_retention
+  kms_key_id        = aws_kms_key.this.arn
+  tags              = var.tags
+}
+
+# Create ECS cluster
+resource "aws_ecs_cluster" "this" {
+  #checkov:skip=CKV_AWS_65:Container Insights will be enabled when required
+  name = var.cluster_name
+
+  setting {
+    name  = "containerInsights"
+    value = var.container_insights
+  }
+
+  configuration {
+    execute_command_configuration {
+      kms_key_id = aws_kms_key.this.arn
+      logging    = "OVERRIDE"
+
+      log_configuration {
+        cloud_watch_encryption_enabled = true
+        cloud_watch_log_group_name     = aws_cloudwatch_log_group.this.name
+      }
+    }
+  }
+  tags = var.tags
+}
+
+# Create ECS IAM Instance Role and Policy
+resource "random_id" "code" {
+  byte_length = 4
+}
+
+resource "aws_iam_role" "ecs_instance_role" {
+  name               = "EcsInstanceRole-${var.cluster_name}-${random_id.code.hex}"
```

Comment:
can you truncate at 64 chars?

Reply:
Yes. I also think I can omit the random code at the end, or use the last 5 character positions for the randomness.